### PR TITLE
Implemented API route to return a specific skill

### PR DESCRIPTION
**Pull request by nfallah**
## Description
Updated the `/resume/skill` endpoint for GET requests to accept index values and return a single experience.

## Issue
This request closes issue #14.

## Testing
An example GET request for `/resume/skill?index=value` will return a valid skill in JSON form if `value` is numeric and within bounds. There are two examples below.

A GET request for `/resume/skill?index=0` will succeed:
```json
{
  "logo":"example-logo.png",
  "name":"Python",
  "proficiency":"1-2 Years"
}
```

A GET request for `/resume/skill?index=1` will fail and return a generic error:
```json
{
  "error":"skill not found"
}
```

### CODE REVIEW

**Review by sudiptob2 — COMMENTED** [review]
In my opinion, we should be using dynamic ULR params for fetching specific data, rathar than using query_params For example, for fetching education with id=1 the route should look like `education/1` rather than `education/index=1` 
See variable rules section in the flask docs https://flask.palletsprojects.com/en/2.3.x/quickstart/#routing
So, for fetching doing operation with specific `skill` there should be another route such as, `skill/<id:int>`

File: app.py [hunk 1]
```diff
@@ -93,8 +93,19 @@ def skill():
...
     if request.method == 'GET':
-        return jsonify({})
+        skills = data.get('skill', [])
+        index = request.args.get('index')
+
+        # Returns a specific skill if the index was specified by the user
+        if index is not None:
+            try:
+                index = int(index)
+                return jsonify(skills[index])
```
[the remark this answers is not in the document]
Comment (Pradyuman7) [reply app.py:103]: What do you think about this @nfallah?

[the remark this answers is not in the document]
Comment (nfallah) [reply app.py]: Agreed -- I have just committed some changes that should make the error-checking more explicit.

**Review by Pradyuman7 — APPROVED** [review]
LGTM

**Pull request merged**